src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -398,142 +340,82 @@ private void failUnpublished(final List<BatchItem> batch, final String reason) {
         try {
             return unmodifiableList(producer.partitionsFor(topicId)
                     .stream()
-                    .map(partitionInfo -> toNakadiPartition(partitionInfo.partition()))
+                    .map(partitionInfo -> KafkaCursor.toNakadiPartition(partitionInfo.partition()))
                     .collect(toList()));
         } finally {
             kafkaFactory.releaseProducer(producer);
         }
     }
 
-    private String transformNewestOffset(final Long newestOffset) {
-        return newestOffset == 0 ? Cursor.BEFORE_OLDEST_OFFSET : toNakadiOffset(newestOffset - 1);
-    }
-
-    private Map<Integer, Long> getPositions(final Consumer<String, String> consumer,
-                                            final List<org.apache.kafka.common.TopicPartition> kafkaTPs) {
-        return kafkaTPs
-                .stream()
-                .collect(Collectors.toMap(
-                        org.apache.kafka.common.TopicPartition::partition,
-                        consumer::position
-                ));
-    }
-
-    @Override
-    public TopicPartition getPartition(final String topicId, final String partition)
```

Comment:
This was optimized to get offsets for a single partition because the seeking is a very time-consuming operation. Now you replaced it with getting offsets for all partitions and selecting a single one. That will lower the performance for GET /event-types/xxx/partitions/XXX